indico/modules/attachments/controllers/event_package.py

```diff
@@ -99,17 +125,39 @@ def _filter_by_date(self, query, added_since):
         return query.filter(cast(AttachmentFile.created_dt, Date) >= added_since)
 
     def _generate_zip_file(self, attachments):
-        temp_file = tempfile.NamedTemporaryFile('w')
+        # XXX: could use a celery task to delay the temporary file after a day or so.
+        # right now this relies on an external cronjob to do so...
+        temp_file = NamedTemporaryFile(suffix='indico.tmp', dir=Config.getInstance().getTempDir(), delete=False)
         with ZipFile(temp_file.name, 'w', allowZip64=True) as zip_handler:
             for attachment in attachments:
-                if not attachment.folder.is_default:
-                    name = os.path.join(os.path.join(secure_filename(attachment.folder.title,
-                                                                     unicode(attachment.folder.id)),
-                                                     attachment.file.filename))
-                else:
-                    name = attachment.file.filename
-
+                name = self._prepare_folder_structure(attachment)
                 with attachment.file.storage.get_local_path(attachment.file.storage_file_id) as filepath:
                     zip_handler.write(filepath, name)
 
         return send_file('material-{}.zip'.format(self._conf.id), temp_file.name, 'application/zip', inline=False)
+
+    def _prepare_folder_structure(self, attachment):
+        event_dir = secure_filename(self._conf.getTitle(), None)
+        path_paths = [event_dir] if event_dir else []
```

Comment:
`path_paths`?! `path_segments` or `path_parts` would be better fits.. or maybe even just `segments`